praw/models/reddit/widgets.py

```diff
@@ -0,0 +1,1397 @@
+"""Provide classes relating to widgets."""
+
+from copy import deepcopy
+from json import dumps
+import re
+import os
+
+from ..base import PRAWBase
+from ...const import API_PATH
+
+
+class Dictable(PRAWBase):
+    """Abstract class to represent a class that can be turned to a dict."""
+
+    def _to_dict(self):
+        self_dict = {}
+        for attr_name, attr_value in self.__dict__.items():
+            if not attr_name.startswith('_'):
+                if isinstance(attr_value, list):
+                    attr_value = [item._to_dict() if isinstance(item, Dictable)
+                                  else item
+                                  for item in attr_value]
+                self_dict[attr_name] = attr_value
+        return self_dict
+
+
+# pylint: disable=no-member
+class Button(Dictable):
+    """Class to represent a single button inside a widget.
+
+    Create a button like so:
+
+    .. code-block:: python
+
+       widgets = reddit.subreddit('redditdev').widgets
+       button = widgets.button('Click me!',
```

Comment:
I didn't think about this. You're right, this makes a lot more sense. Viewing widgets can be done by anybody, but editing or creating them is a mod action. Because of this, all moderator-only methods should be separated out just like in the rest of PRAW.